File: java/src/entity/Membre.java
package entity;

import utils.Date;

public class Membre {
    private Date dateIntegration;
    private Groupe groupe;
    private User user;
    private TypeMembre role;

    public Membre(Date dateIntegration, Groupe groupe, User user) {
        this.dateIntegration = dateIntegration;
        this.groupe = groupe;
        this.user = user;
        this.role = TypeMembre.MEMBRE;
    }

    // Getters
    public  Date getDateIntegration() { return dateIntegration; }
    public Groupe getGroupe() 
    { 
        return groupe; 
    }
    public User getUser() { return user; }
    public TypeMembre getRole() { return role; }

    // Setters
    public void setDateIntegration(Date dateIntegration) { this.dateIntegration = dateIntegration; }
    public void  setGroupe(Groupe groupe) { this.groupe = groupe; }
    public void setUser(User user) { this.user = user; }
    public void setTypeMembre(TypeMembre role) { this.role = role; }
}

File: php/src/entity/Membre.php
<?php

namespace App\Entity;

use App\Entity\TypeMembre;
use DateTime;

class Membre
{
    private int $id;
    private int $groupe_id;
    private int $user_id;
    private string $integreted_at;
    private TypeMembre|string $role;
    // ATTRIBUTS METIERS
    private Groupe $groupe;
    private User $user;

    //CONSTRUCTEUR
    public function __construct() {}

    // GETTERS
    public function getDateIntegration()
    {
        return $this->integreted_at;
    }
    public function getGroupeId(): int
    {
        return $this->groupe_id;
    }
    public function getUserId(): int
    {
        return $this->user_id;
    }
    public function getGroupe(): Groupe
    {
        return $this->groupe;
    }
    public function getUser(): User
    {
        return $this->user;
    }
    public function getRole(): TypeMembre
    {
        if (is_string($this->role)) return TypeMembre::from($this->role);
        return $this->role;
    }

    // SETTERS
    public function  setGroupeId(int $groupe_id)
    {
        $this->groupe_id = $groupe_id;
    }
    public function setUserId(int $user_id)
    {
        $this->user_id = $user_id;
    }
    public function setTypeMembre(TypeMembre $role)
    {
        $this->role = $role;
    }
    public function  setGroupe(Groupe $groupe)
    {
        $this->groupe = $groupe;
    }
    public function setUser(User $user)
    {
        $this->user = $user;
    }

    public function toDbArray(): array
    {
        return [
            "groupe_id" => $this->groupe_id,
            "user_id" => $this->user_id,
            "role" => $this->role instanceof TypeMembre ? $this->role->value : $this->role,
        ];
    }
}
